### ISSUE TITLE
help() does not work correctly (Something odd with ClassProperty)

### ISSUE THREAD

**Issue by ecobost** [datajoint/datajoint-python#306]
```python
In [1]: import datajoint as dj
DataJoint 0.4.10 (February 6, 2017)
No config file found, using default settings.

In [2]: help(dj.Computed)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-4915eceda6d7> in <module>()
----> 1 help(dj.Computed)

/usr/lib/python3.5/_sitebuiltins.py in __call__(self, *args, **kwds)
    101     def __call__(self, *args, **kwds):
    102         import pydoc
--> 103         return pydoc.help(*args, **kwds)

/usr/lib/python3.5/pydoc.py in __call__(self, request)
   1850     def __call__(self, request=_GoInteractive):
   1851         if request is not self._GoInteractive:
-> 1852             self.help(request)
   1853         else:
   1854             self.intro()

/usr/lib/python3.5/pydoc.py in help(self, request)
   1903             else: doc(str, 'Help on %s:', output=self._output)
   1904         elif isinstance(request, Helper): self()
-> 1905         else: doc(request, 'Help on %s:', output=self._output)
   1906         self.output.write('\n')
   1907 

/usr/lib/python3.5/pydoc.py in doc(thing, title, forceload, output)
   1637     try:
   1638         if output is None:
-> 1639             pager(render_doc(thing, title, forceload))
   1640         else:
   1641             output.write(render_doc(thing, title, forceload, plaintext))

/usr/lib/python3.5/pydoc.py in render_doc(thing, title, forceload, renderer)
   1630         object = type(object)
   1631         desc += ' object'
-> 1632     return title % desc + '\n\n' + renderer.document(object, name)
   1633 
   1634 def doc(thing, title='Python Library Documentation: %s', forceload=0,

/usr/lib/python3.5/pydoc.py in document(self, object, name, *args)
    373         try:
    374             if inspect.ismodule(object): return self.docmodule(*args)
--> 375             if inspect.isclass(object): return self.docclass(*args)
    376             if inspect.isroutine(object): return self.docroutine(*args)
    377         except AttributeError:

/usr/lib/python3.5/pydoc.py in docclass(self, object, name, mod, *ignored)
   1274 
   1275         attrs = [(name, kind, cls, value)
-> 1276                  for name, kind, cls, value in classify_class_attrs(object)
   1277                  if visiblename(name, obj=object)]
   1278 

/usr/lib/python3.5/pydoc.py in classify_class_attrs(object)
    208     """Wrap inspect.classify_class_attrs, with fixup for data descriptors."""
    209     results = []
--> 210     for (name, kind, cls, value) in inspect.classify_class_attrs(object):
    211         if inspect.isdatadescriptor(value):
    212             kind = 'data descriptor'

/usr/lib/python3.5/inspect.py in classify_class_attrs(cls)
    395                     # first look in the classes
    396                     for srch_cls in class_bases:
--> 397                         srch_obj = getattr(srch_cls, name, None)
    398                         if srch_obj is get_obj:
    399                             last_cls = srch_cls

/usr/local/lib/python3.5/dist-packages/datajoint-0.4.10-py3.5.egg/datajoint/utils.py in __get__(self, obj, owner)
      8 
      9     def __get__(self, obj, owner):
---> 10         return self.f(owner)
     11 
     12 

/usr/local/lib/python3.5/dist-packages/datajoint-0.4.10-py3.5.egg/datajoint/user_relations.py in table_name(cls)
     62         :returns: the table name of the table formatted for mysql.
     63         """
---> 64         return cls._prefix + from_camel_case(cls.__name__)
     65 
     66     @ClassProperty

TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'

In [3]: dj.Computed?
Type:            OrderedClass
String form:     <class 'datajoint.user_relations.Computed'>
File:            /usr/local/lib/python3.5/dist-packages/datajoint-0.4.10-py3.5.egg/datajoint/user_relations.py
Init definition: dj.Computed(self, arg=None)
Docstring:
Inherit from this class if the table's values are computed from other relations in the schema.
The inherited class must at least provide the function `_make_tuples`.

In [4]: dj.Computed.table_name
Out[4]: '__computed'
```
help(dj) fails with the same exception too. I believe it has to do with the ClassProperty decorator but from the code it looks like it should work. I believe someone (pydoc) is calling UserRelation.table_name and as UserRelation._prefix = None it breaks.  I also don't know whether this matters, if it is only pydoc that breaks maybe it is not that important.

**Comment by eywalker** [comment]
Ultimately, this is caused by the fact that `pydoc` doesn't expect that anything should be triggered by referring to the class property - indeed, there is no such thing as class level property from Python and we have implemented this special decorator ourselves. Fortunately for this case, the fix seems to be easy.

**Comment by eywalker** [comment]
This issue was addressed in PR #307. However, let's take note of the fact that the `classproperty` we use is something we wrote and thus can break expectation that some other libraries might have about class properties (i.e. in this case, the pydoc module didn't expect that accessing a property on class can invoke a method call). In principle, there is no problem setting class property to be a [descriptor](https://docs.python.org/3.6/howto/descriptor.html) but it is definitely not a common practice.
